soc/xlnx/versalnet/soc.c

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2025 Advanced Micro Devices, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/cache.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+
+void z_arm_platform_init(void)
```

Comment:
In order for z_arm_platform_init() to be executed, you'll have to select PLATFORM_SPECIFIC_INIT. However, this is deprecated and therefore shouldn't be used. You can do things like disabling the caches, adjusting VBAR if available, or making sure that HIVECS is cleared at the earliest stages of booting in soc_reset_hook() which goes into soc.c. With this current implementation of z_arm_platform_init(), the caches are enabled before the MPU is set up.
Have a look at #85537 as reference, this PR is currently in progress, adding cache support for the Cortex-R5.

Reply:
sure fill fix